src/script/conversation/ConversationVerificationStateHandler/shared/conversation/index.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {QualifiedId} from '@wireapp/api-client/lib/user';
+
+import {ConversationState} from 'src/script/conversation/ConversationState';
+import {ConversationVerificationState} from 'src/script/conversation/ConversationVerificationState';
+import {Conversation} from 'src/script/entity/Conversation';
+import {UserState} from 'src/script/user/UserState';
+import {Logger} from 'Util/Logger';
+import {matchQualifiedIds} from 'Util/QualifiedId';
+
+interface GetActiveConversationsWithUsersParams {
+  userIds: QualifiedId[];
+  conversationState: ConversationState;
+  userState: UserState;
+}
+type GetActiveConversationsWithUsersResult = {conversationEntity: Conversation; userIds: QualifiedId[]}[];
+
+export const getActiveConversationsWithUsers = ({
+  conversationState,
+  userIds,
+  userState,
+}: GetActiveConversationsWithUsersParams): GetActiveConversationsWithUsersResult => {
+  return conversationState
+    .filteredConversations()
+    .map((conversationEntity: Conversation) => {
+      if (!conversationEntity.removed_from_conversation()) {
+        const userIdsInConversation = conversationEntity.participating_user_ids().concat(userState.self().qualifiedId);
+        const matchingUserIds = userIdsInConversation.filter(userIdInConversation =>
+          userIds.find(userId => matchQualifiedIds(userId, userIdInConversation)),
+        );
+
+        if (!!matchingUserIds.length) {
+          return {conversationEntity, userIds: matchingUserIds};
+        }
+      }
+      return undefined;
+    })
+    .filter(activeConversationInfo => !!activeConversationInfo) as {
```

Comment:
i always prefer this style but this is just a suggestion:

```suggestion
    .filter(Boolean) as {
```